pkg/fetcher/errors.go

```diff
@@ -43,3 +46,13 @@ type TagNotFoundError struct {
 func (e TagNotFoundError) Error() string {
 	return fmt.Sprintf("image tag not found: %s", e.Err.Error())
 }
+
+// AuthTokenError is returned when authentication with a registry occurs
+type AuthTokenError struct {
+	TokenServer url.URL
+	Err         error
+}
+
+func (e AuthTokenError) Error() string {
+	return fmt.Sprintf("Failed to fetch auth token from %s", e.TokenServer.Host)
```

Comment:
If we're just using the `TokenServer.Host` field, can we store just the `Host` in the `AuthTokenError` struct instead of the entire URL?